packages/horizontal-slider-react/src/HorizontalSlider.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { useEffect, useMemo, useRef } from 'react';
+
+let lastHorizontalSliderId = 0;
+
+function generateHorizontalSliderId() {
+  return `uic-horizontal-slider-${lastHorizontalSliderId++}`;
+}
+
+function cx(...classNames: Array<string | undefined>) {
+  return classNames.filter(Boolean).join(' ');
+}
+
+type RecordWithObjectID<TObject = {}> = TObject & {
+  objectID: string;
+};
+
+type HorizontalSliderClassnames = {
+  item: string;
+  list: string;
+  navigation: string;
+  navigationNext: string;
+  navigationPrevious: string;
+  root: string;
+};
+
+type HorizontalSliderTranslations = {
+  sliderLabel: string;
+  previousButtonLabel: string;
+  previousButtonTitle: string;
+  nextButtonLabel: string;
+  nextButtonTitle: string;
+};
+
+export type HorizontalSliderProps<TItem extends RecordWithObjectID> = {
+  items: TItem[];
+  itemComponent({ item: TItem }): JSX.Element;
+  classNames?: Partial<HorizontalSliderClassnames>;
+  translations?: Partial<HorizontalSliderTranslations>;
+};
+
+export function HorizontalSlider<TObject extends RecordWithObjectID>(
+  props: HorizontalSliderProps<TObject>
+) {
+  const translations: HorizontalSliderTranslations = useMemo(
+    () => ({
+      sliderLabel: 'Items',
+      nextButtonLabel: 'Next',
+      nextButtonTitle: 'Next',
+      previousButtonLabel: 'Previous',
+      previousButtonTitle: 'Previous',
+      ...props.translations,
+    }),
+    [props.translations]
+  );
+  const listRef = useRef<HTMLOListElement>(null);
+  const previousButtonRef = useRef<HTMLButtonElement>(null);
+  const nextButtonRef = useRef<HTMLButtonElement>(null);
+  const sliderIdRef = useRef(generateHorizontalSliderId());
+
+  function scrollLeft() {
+    if (listRef.current) {
+      listRef.current.scrollLeft -= listRef.current.offsetWidth * 0.75;
+    }
+  }
+
+  function scrollRight() {
+    if (listRef.current) {
+      listRef.current.scrollLeft += listRef.current.offsetWidth * 0.75;
+    }
+  }
+
+  function updateNavigationButtonsProps() {
+    if (
+      !listRef.current ||
+      !previousButtonRef.current ||
+      !nextButtonRef.current
+    ) {
+      return;
+    }
+
+    previousButtonRef.current.hidden = listRef.current.scrollLeft <= 0;
+    nextButtonRef.current.hidden =
+      listRef.current.scrollLeft + listRef.current.clientWidth >=
+      listRef.current.scrollWidth;
+  }
+
+  useEffect(() => {
+    updateNavigationButtonsProps();
+  });
+
+  if (props.items.length === 0) {
+    return null;
+  }
+
+  return (
+    <div
+      className={cx('uic-HorizontalSlider-container', props.classNames?.root)}
+    >
+      <button
+        ref={previousButtonRef}
+        title={translations.previousButtonTitle}
+        aria-label={translations.previousButtonLabel}
+        aria-controls={sliderIdRef.current}
+        className={cx(
+          'uic-HorizontalSlider-navigation',
+          'uic-HorizontalSlider-navigation--previous',
+          props.classNames?.navigation,
```

Comment:
maybe classnames should be defaulted to {} to avoid checking at every usage?

Reply:
See 4b843ec6ba26b055ee4477af7f65aabd04590576.